packages/flutter/test/painting/decoration_image_lerp_test.dart

```diff
@@ -579,21 +586,40 @@ void main() {
       find.byType(Wrap),
       matchesGoldenFile('decoration_image.lerp.2.png'),
     );
+
+    cmykLoader.dispose();
+    wrgbLoader.dispose();
+    imageCache.clear();
   }, skip: kIsWeb); // TODO(ianh): https://github.com/flutter/flutter/issues/130612, https://github.com/flutter/flutter/issues/130609
 }
 
-Future<void> load(MemoryImage image) {
-  final ImageStream stream = image.resolve(ImageConfiguration.empty);
-  final Completer<ImageInfo> completer = Completer<ImageInfo>();
-  void listener(ImageInfo image, bool syncCall) {
-    completer.complete(image);
+class _ImageLoader {
+  _ImageLoader(this.image);
+
+  late final MemoryImage image;
+  late final ImageStream stream;
+  final Completer<ImageInfo> _completer = Completer<ImageInfo>();
+  late final ImageStreamListener wrappedListener;
+
+  void _listener(ImageInfo image, bool syncCall) {
+    _completer.complete(image);
+    addTearDown(image.dispose);
+  }
+
+  Future<void> load() {
+    stream = image.resolve(ImageConfiguration.empty);
+    wrappedListener = ImageStreamListener(_listener);
+    stream.addListener(wrappedListener);
+    return _completer.future;
+  }
+
+  void dispose(){
+    stream.removeListener(wrappedListener);
   }
-  stream.addListener(ImageStreamListener(listener));
-  return completer.future;
 }
 
-class TestImage extends StatelessWidget {
-  TestImage(this.image); // ignore: use_key_in_widget_constructors, prefer_const_constructors_in_immutables
+class _TestImage extends StatelessWidget {
+  _TestImage(this.image); // ignore: use_key_in_widget_constructors, prefer_const_constructors_in_immutables
```

Comment:
I believe the ignore for use_key_in_widget_constructors is no longer needed since this is a private class now.